aea/multiplexer.py

```diff
@@ -699,6 +702,21 @@ def _setup(
         for c in connections:
             self.add_connection(c, c.public_id == default_connection)
 
+    @staticmethod
+    def _set_context_on_envelope(envelope: Envelope, connection: Connection) -> None:
+        """
+        Updat the envelope context.
+
```

Comment:
also, if you don't mind, would be great to have the explanation in the docstring of the three cases below.